tests/Book/RefreshTest.php

```diff
@@ -9,39 +9,48 @@
 use GuzzleHttp\Handler\MockHandler;
 use GuzzleHttp\HandlerStack;
 use GuzzleHttp\Psr7\Response;
-use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
 /**
  * @covers Refresh
  */
-class RefreshTest extends TestCase {
+class RefreshTest extends KernelTestCase {
+
+	/** @var Api */
+	private $api;
+
+	public function setUp(): void {
+		parent::setUp();
+		self::bootKernel();
+		$this->api = self::$kernel->getContainer()->get( Api::class );
```

Comment:
With the following you can access private services in tests so you don't need to make them public
```suggestion
		$this->api = self::$container->get( Api::class );
```

Reply:
thanks! I figured I was missing something clever. :) Done.

Reply:
Actually I'm not sure this makes any difference. I'm still getting "Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The "App\Util\Api" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead."